TransformationSystem/Client/TaskManager.py

```diff
@@ -258,23 +258,19 @@ def getSubmittedFileStatus( self, fileDicts ):
       transID = fileDict['TransformationID']
       taskID = int( fileDict['TaskID'] )
       if taskID in submittedTasks[transID]:
-        requestID = externalIds[taskID]
-        taskFiles.setdefault( requestID, {} )[fileDict['LFN']] = fileDict['Status']
+        taskFiles.setdefault( externalIds[taskID], [] ).append( fileDict['LFN'] )
 
     updateDict = {}
     for requestID in sorted( taskFiles ):
-      lfnDict = taskFiles[requestID]
-      statusDict = self.requestClient.getRequestFileStatus( requestID, lfnDict.keys() )
+      lfnList = taskFiles[requestID]
+      statusDict = self.requestClient.getRequestFileStatus( requestID, lfnList )
       if not statusDict['OK']:
         log = self._logVerbose if 'not exist' in statusDict['Message'] else self.log.warn
         log( "getSubmittedFileStatus: Failed to get files status for request", '%s' % statusDict['Message'] )
         continue
 
-      statusDict = statusDict['Value']
-      for lfn, newStatus in statusDict.items():
-        if newStatus == lfnDict[lfn]:
-          pass
```

Comment:
Because it is comparing apples with pears! The name of the statuses are different and there is no reasons why they would match!
newStatus is the status of the file in the request (e.g. Waiting, Scheduled, Done)
lfnDict[lfn] is the status of the file in the TS (e.g. Unused, Assigned, Processed)
;-)))